### PR TITLE
Issue 1388

### PR DESCRIPTION
**Pull request by ciaranlp**
add the ability to skip to menu or page main content bypassing the navigation links

https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-skip.html

### CODE REVIEW

File: src/core/js/router.js [hunk 1]
```diff
@@ -206,6 +207,20 @@ define([
...
+            var pageCount = $(".page").length;
+            var pageContent = $(".page-inner");
+            var menuContent = $(".menu");
+
+            if (pageCount == 1) {
+                pageContent.a11y_focus();
+            } else {
+                menuContent.a11y_focus();
+            }
+
```
Comment (oliverfoster) [inline src/core/js/router.js]: this can be replaced with:

```
$("."+Adapt.location._contentType).a11y_focus();
```

Comment (oliverfoster) [inline src/core/js/router.js]: core template also needs updating as per pr to vanilla https://github.com/adaptlearning/adapt-contrib-vanilla/pull/201

File: src/core/templates/navigation.hbs [hunk 2]
```diff
@@ -1,5 +1,8 @@
 {{! All links in the nav bar should have an attritbue of data-event, navigationView uses this to fire a global event}}
 <div class="navigation-inner clearfix" role="navigation" aria-label="{{_globals._accessibility._ariaLabels.navigation}}">
+    {{#if _accessibility._isSkipNavigationEnabled}}
+        <button class="skip-nav-link" aria-label="Skip navigation" data-event="skipNavigation" role="button">Skip Navigation</button>
```
Comment (brian-learningpool) [inline src/core/templates/navigation.hbs]: It doesn't seem right that "Skip Navigation" is hard-coded twice here.  

We need to add a language string to the `_globals._accesibility` object, like `_accessibilityToggleTextOn` and another for the aria-label.

File: src/core/schema/course.model.schema [hunk 3]
```diff
@@ -171,6 +171,17 @@
               ],
               "translatable": true
             },
+            "_skipNavigationText": {
```
Comment (tomgreenfield) [inline src/core/schema/course.model.schema]: Please remove underscore from `_skipNavigationText`.

File: src/core/templates/navigation.hbs [hunk 4]
```diff
@@ -1,5 +1,8 @@
 {{! All links in the nav bar should have an attritbue of data-event, navigationView uses this to fire a global event}}
 <div class="navigation-inner clearfix" role="navigation" aria-label="{{_globals._accessibility._ariaLabels.navigation}}">
+    {{#if _accessibility._isSkipNavigationEnabled}}
+        <button class="skip-nav-link" data-event="skipNavigation" aria-label="{{_globals._accessibility._ariaLabels.skipNavigation}}">{{_globals._accessibility._skipNavigationText}}</button>
```
Comment (tomgreenfield) [inline src/core/templates/navigation.hbs]: Please remove underscore from `_skipNavigationText`.

Comment (tomgreenfield) [inline src/core/templates/navigation.hbs]: Sorry I missed this the first time around but we usually allow HTML with triple braces `{{{` `}}}` on the actual rendered text e.g. on [Box Menu buttons](https://github.com/adaptlearning/adapt-contrib-boxmenu/blob/master/templates/boxmenu-item.hbs#L15).

**Comment by moloko** [comment]
sorry all, I've only just spotted this:

If we're going to have `_isSkipNavigationEnabled` enabled by default (which it currently is in both src/course/config.json and config.model.schema) then we also need to make sure there's a `skipNavigation` ARIA label in the default course.json.

Or we don't have it enabled by default in which case it doesn't matter so much if the ARIA label isn't present.

I don't mind which we go with.

**Comment by ciaranlp** [comment]
hey guys a PR for @moloko last comment can be found here https://github.com/adaptlearning/adapt_framework/pull/1414

**Pull request merged**